dgisim/src/dices.py

```diff
@@ -241,6 +272,206 @@ def just_satisfy(self, requirement: AbstractDices) -> bool:
             return False
         return self._satisfy(requirement)
 
+    @staticmethod
+    def _how_much_omni_and_first_priority_to_spend(*,
+                                                   omni_supply: int,
+                                                   omni_needed: int,
+                                                   element_supply: int,
+                                                   element_needed: int,
+                                                   element_prioritized: bool
+                                                   ) -> int:
+        """
+        how many OMNI (+ first priority elements if it so) spend if we select this ELEMENT as OMNI filler.
+        return BIG_INT if no possibility
+        """
+        if element_supply + omni_supply >= element_needed + omni_needed:
+            if element_prioritized:
+                return element_needed + omni_needed
+            else:
+                return max((element_needed + omni_needed) - element_supply, 0)
+        else:
+            # return None
+            return BIG_INT
```

Comment:
Is it possible to move the `else: return BIG_INT` part to the list comprehension where this method is called, to be like:

```py
some_list = [
    (...)
    for ... in ...
    if ...  # move condition here
]

Without this _outer condition_, the inner one isn't that long.

You may either leave the inner `if` statement as it is (but move the method inside the `smart_selection()`; or inline the `if` statement to the list-comprehension where you used this method.

Reply:
fixed, if I understand you it right.